directory/directory.go

```diff
@@ -37,7 +38,15 @@ func NewDirectory(cfg *ds.Config) (*Directory, error) {
 		return nil, errors.Wrap(err, "failed to create directory client")
 	}
 
-	return &Directory{Client: client}, nil
+	isLegacy, err := isLegacy(client)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to determine if directory is legacy")
```

Comment:
I would use a more descriptive error message. Something like:
`failed to determine direction of 'identifier' relation`